types/site/_utils/add-pagination.d.ts

```diff
@@ -15,13 +15,12 @@
  */
 
 declare global {
-  export interface PaginatedPage<T = EleventyCollectionItem> {
+  export interface PaginatedPage extends VirtualCollectionItem {
```

Comment:
I still think this `extends` doesn't make total sense but I think we're getting closer here.

My understanding is that:

- e.g. `AuthorsItem` inherits from `VirtualCollectionItem`, and has a property `.elements` which is of type `FrontMatterData[]`
  - This represents the author's posts, which makes sense—those map to posts on-disk, which are represented by `FrontMatterData` (...& you've added valid URLs to the front-matter)

- Every `PaginatedPage` basically ends up looking more or less the same as the source `AuthorsItem` (because you apply its contents via `...`, and this is why inheriting from `VirtualCollectionItem` makes sense), but with elements sliced up _and_ the `paginatined` property added

(1) I still think we should put the underlying type on a `.item` property here, and get rid of the extends (but I could be convinced otherwise)
(2) the pagination object _should not_ actually be an `EleventyPagination`, since ... it's not. But I appreciate that this was here before and my PR fixes that.